fontc/src/config.rs

```diff
@@ -0,0 +1,133 @@
+//! State for a (possibly incremental) compiler job
+
+use std::{
+    fs, io,
+    path::{Path, PathBuf},
+};
+
+use fontir::{paths::Paths as IrPaths, source::Input, stateset::StateSet};
+use serde::{Deserialize, Serialize};
+
+use crate::Args;
+
+/// The settings and compiler definition of a single compilation run.
+///
+/// This tracks the input arguments for a compilation, as well as the compiler
+/// version used (so that we do not reuse incremental state between different
+/// compiler versions)
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq)]
+pub struct Config {
+    pub args: Args,
+    // The compiler previously used so if the compiler changes config invalidates
+    compiler: StateSet,
+}
+
+impl Config {
+    /// Create a new config from the provided cli arguments
+    pub fn new(args: Args) -> Result<Config, io::Error> {
+        let mut compiler = StateSet::new();
+        compiler.track_file(&std::env::current_exe()?)?;
+        Ok(Config { args, compiler })
+    }
+
+    /// Returns the path to the config file for this compilation
+    fn file(&self) -> PathBuf {
+        self.args.build_dir.join("fontc.yml")
+    }
+
+    /// Attempt to initialize the compiler for this run.
+    ///
+    /// This serializes the configuration if necessary, and returns any existing
+    /// incremental state for this run if it exists.
+    //FIXME: this returns a Result but it looks like we never actually return
```

Comment:
I think that's an error and we should change the expect's to map_err or ?. Lets do that in it's own PR as it's hard to see diffs when mixed with a move.